src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1887,6 +2006,27 @@ def update_nat_gateway_profile(self, mc: ManagedCluster) -> ManagedCluster:
             )
         return mc
 
+    def update_pod_identity_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update pod identity profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_pod_identity():
```

Comment:
question: what's the differences between `self._get_disable/enable_pod_identity(enable_validation=True)` and `context. get_enable_pod_identity` here?

Reply:
Briefly, no difference.

But I need to remind that both functions are defined in the context class (e.g. `AKSContext`/`AKSPreviewContext`), and to call them in the decorator class (e.g. `AKSCreateDecorator`/`AKSPreviewCreateDecorator`), it should be `self.context._get_enable_pod_identity(enable_validation=True)  # not recommended` or `self.context.get_enable_pod_identity()`.